podcasts/EpisodesDataManager.swift

```diff
@@ -3,6 +3,30 @@ import PocketCastsServer
 import DifferenceKit
 
 class EpisodesDataManager {
+    // MARK: - Playlist episodes
+
+    /// Return the list of episodes for a given playlist
+    func episodes(for playlist: AutoplayHelper.Playlist) -> [BaseEpisode] {
+        switch playlist {
+        case .podcast(uuid: let uuid):
+            if let podcast = DataManager.sharedManager.findPodcast(uuid: uuid, includeUnsubscribed: true) {
+                return Array(episodes(for: podcast).compactMap { $0.elements as? [ListEpisode] }.joined()).map { $0.episode }
+            }
+        case .filter(uuid: let uuid):
+            if let filter = DataManager.sharedManager.findFilter(uuid: uuid) {
+                return episodes(for: filter).compactMap { $0.episode }
```

Comment:
[nit] since `episodes(for: filter)` is returning an array without any nil items we can just use `map` here. 


```suggestion
                return episodes(for: filter).map { $0.episode }
```

Reply:
.map is also slightly faster when processing lots of items than .compactMap

<img width="516" alt="Screenshot 2023-06-29 at 4 33 42 PM" src="https://github.com/Automattic/pocket-casts-ios/assets/793774/604fc3e4-ab2d-4056-8e52-614869ab281c">